### PR TITLE
Enable some mandatory `VK_EXT_descriptor_indexing` features.

### PR DESCRIPTION
**Pull request by cdavis5e**
These features must be supported if the extension or `descriptorIndexing` feature is.

Fixes `dEQP-VK.info.device_mandatory_features`.

### CODE REVIEW

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm [hunk 1]
```diff
@@ -126,14 +126,14 @@
 		.shaderInt8 = true,
 		.descriptorIndexing = _vulkan12FeaturesNoExt.descriptorIndexing,
 		.shaderInputAttachmentArrayDynamicIndexing = _metalFeatures.arrayOfTextures,
-		.shaderUniformTexelBufferArrayDynamicIndexing = _metalFeatures.arrayOfTextures,
```
Comment (billhollings) [inline MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:129]: When does `_metalFeatures.arrayOfTextures` fail? Are you running CTS on an ancient devices or OS?

Perhaps we should disable the `VK_EXT_descriptor_indexing` extension completely in these cases. Perhaps test `_metalFeatures.arrayOfTextures` in `MVKPhysicalDevice::initExtensions()`.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm [hunk 2]
```diff
@@ -126,14 +126,14 @@
...
-		.shaderStorageTexelBufferArrayDynamicIndexing = _metalFeatures.arrayOfTextures,
-		.shaderUniformBufferArrayNonUniformIndexing = false,
```
Comment (billhollings) [inline MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:131]: What are the ramifications of indicating such complete support?

Will it cause other CTS tests, that check these flags, to fail if we actually don't support them?

**Review by billhollings — COMMENTED** [review]
A couple of inline questions.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm (hunk 1 above)
Comment (cdavis5e) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:129]: I had thought that we were enabling `shaderSampledImageDynamicIndexing` unconditionally, but, on closer inspection, we weren't. I've changed it to disable the extension if the requisite Metal features aren't supported, as you suggested.

File: MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm (hunk 2 above)
Comment (billhollings) [reply MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm:131]: > I don't see why non-uniform indexing can't work.

Ah. Good point.

**Review by billhollings — APPROVED** [review]

**Pull request merged**